dids/didcore/document.go

```diff
@@ -83,20 +95,101 @@ func (d *Document) AddVerificationMethod(method VerificationMethod, opts ...AddV
 
 	for _, p := range o.purposes {
 		switch p {
-		case "assertionMethod":
+		case PurposeAssertion:
 			d.AssertionMethod = append(d.AssertionMethod, method.ID)
-		case "authentication":
+		case PurposeAuthentication:
 			d.Authentication = append(d.Authentication, method.ID)
-		case "keyAgreement":
+		case PurposeKeyAgreement:
 			d.KeyAgreement = append(d.KeyAgreement, method.ID)
-		case "capabilityDelegation":
+		case PurposeCapabilityDelegation:
 			d.CapabilityDelegation = append(d.CapabilityDelegation, method.ID)
-		case "capabilityInvocation":
+		case PurposeCapabilityInvocation:
 			d.CapabilityInvocation = append(d.CapabilityInvocation, method.ID)
 		}
 	}
 }
 
+// VMSelector is an interface that can be implemented to provide a means to select
+// a specific verification method from a DID Document.
+type VMSelector interface {
+	selector()
```

Comment:
I might be missing something. is this method itself used anywhere? or mostly just to enforce that the types that fit VMSelector are defined here. (aka Purpose and ID)

Reply:
if that's the case, very cool.